File: Examples/FragmentStaticConfigLayout/src/course/examples/Fragments/StaticConfigLayout/TitlesFragment.java
package course.examples.Fragments.StaticConfigLayout;

import android.app.Activity;
import android.app.ListFragment;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ListView;

//Several Activity and Fragment lifecycle methods are instrumented to emit LogCat output
//so you can follow the class' lifecycle

public class TitlesFragment extends ListFragment {

	private static final String TAG = "TitlesFragment";
	private ListSelectionListener mListener = null;
	private int mCurrIdx = -1;

	// Callback interface that allows this Fragment to notify the QuoteViewerActivity when  
	// user clicks on a List Item  
	public interface ListSelectionListener {
		public void onListSelection(int index);
	}

	@Override
	public void onAttach(Activity activity) {
		super.onAttach(activity);
		try {
			
			// Set the ListSelectionListener for communicating with the QuoteViewerActivity
			mListener = (ListSelectionListener) activity;

		} catch (ClassCastException e) {
			throw new ClassCastException(activity.toString()
					+ " must implement OnArticleSelectedListener");
		}
	}

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		setRetainInstance(true);
	}

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container,
			Bundle savedInstanceState) {
		return super.onCreateView(inflater, container, savedInstanceState);

	}

	@Override
	public void onActivityCreated(Bundle savedState) {
		super.onActivityCreated(savedState);

		// Set the list choice mode to allow only one selection at a time
		getListView().setChoiceMode(ListView.CHOICE_MODE_SINGLE);

		// Set the list adapter for the ListView 
		// Discussed in more detail in the user interface classes lesson  
		setListAdapter(new ArrayAdapter<String>(getActivity(),
				R.layout.list_item, QuoteViewerActivity.mTitleArray));

		// If an item has been selected, set its checked state
		if (-1 != mCurrIdx)
			getListView().setItemChecked(mCurrIdx, true);
	}

	// Called when the user selects an item from the List
	@Override
	public void onListItemClick(ListView l, View v, int pos, long id) {
		if (mCurrIdx != pos) {
			mCurrIdx = pos;
			
			// Inform the QuoteViewerActivity that the item in position pos has been selected
			mListener.onListSelection(pos);
		}
		// Indicates the selected item has been checked
		l.setItemChecked(mCurrIdx, true);
	}

	@Override
	public void onDestroy() {
		Log.i(TAG, getClass().getSimpleName() + ":onDestroy()");
		super.onDestroy();
	}

	@Override
	public void onDestroyView() {
		Log.i(TAG, getClass().getSimpleName() + ":onDestroyView()");
		super.onDestroyView();
	}

	@Override
	public void onDetach() {
		Log.i(TAG, getClass().getSimpleName() + ":onDetach()");
		super.onDetach();
	}

	@Override
	public void onPause() {
		Log.i(TAG, getClass().getSimpleName() + ":onPause()");
		super.onPause();
	}

	@Override
	public void onResume() {
		Log.i(TAG, getClass().getSimpleName() + ":onResume()");
		super.onResume();
	}

	@Override
	public void onStart() {
		Log.i(TAG, getClass().getSimpleName() + ":onStart()");
		super.onStart();
	}

	@Override
	public void onStop() {
		Log.i(TAG, getClass().getSimpleName() + ":onStop()");
		super.onStop();
	}
}

File: ExamplesKotlin/FragmentStaticConfigLayout/app/src/main/java/course/examples/fragments/staticconfiglayout/TitlesFragment.kt
package course.examples.fragments.staticconfiglayout

import android.content.Context
import android.os.Bundle
import android.support.v4.app.ListFragment
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ListView

//Several Activity and Fragment lifecycle methods are instrumented to emit LogCat output
//so you can follow the class' lifecycle

class TitlesFragment : ListFragment() {

    companion object {

        private const val TAG = "TitlesFragment"
    }

    private var mListener: ListSelectionListener? = null
    private var mCurrIdx = ListView.INVALID_POSITION


    override fun onAttach(context: Context) {
        super.onAttach(context)
        Log.i(TAG, javaClass.simpleName + ":onAttach()")

        // Set the ListSelectionListener for communicating with the QuoteViewerActivity
        mListener = context as? ListSelectionListener
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        Log.i(TAG, javaClass.simpleName + ":onCreate()")

        retainInstance = true
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        Log.i(TAG, javaClass.simpleName + ":onCreateView()")

        return super.onCreateView(inflater, container, savedInstanceState)

    }

    override fun onActivityCreated(savedState: Bundle?) {
        super.onActivityCreated(savedState)
        Log.i(TAG, javaClass.simpleName + ":onActivityCreated()")

        // Set the list choice mode to allow only one selection at a time
        listView.choiceMode = ListView.CHOICE_MODE_SINGLE

        // Set the list adapter for the ListView
        // Discussed in more detail in the user interface classes lesson
        listAdapter = ArrayAdapter(
            activity as Context,
            R.layout.list_item, QuoteViewerActivity.mTitleArray
        )
    }

    // Called when the user selects an item from the List
    override fun onListItemClick(l: ListView, v: View, pos: Int, id: Long) {
        if (mCurrIdx != pos) {
            mCurrIdx = pos
            // Indicates the selected item has been checked
            l.setItemChecked(mCurrIdx, true)
            // Inform the QuoteViewerActivity that the item in position pos has been selected
            mListener?.onListSelection(mCurrIdx)
        }
    }

    override fun onDestroy() {
        Log.i(TAG, javaClass.simpleName + ":onDestroy()")
        super.onDestroy()
    }

    override fun onDestroyView() {
        Log.i(TAG, javaClass.simpleName + ":onDestroyView()")
        super.onDestroyView()
    }

    override fun onDetach() {
        Log.i(TAG, javaClass.simpleName + ":onDetach()")
        super.onDetach()
    }

    override fun onPause() {
        Log.i(TAG, javaClass.simpleName + ":onPause()")
        super.onPause()
    }

    override fun onResume() {
        Log.i(TAG, javaClass.simpleName + ":onResume()")
        super.onResume()
    }

    override fun onStart() {
        Log.i(TAG, javaClass.simpleName + ":onStart()")
        super.onStart()
    }

    override fun onStop() {
        Log.i(TAG, javaClass.simpleName + ":onStop()")
        super.onStop()
    }
}